src/index.tsx

```diff
@@ -196,18 +215,23 @@ class PyflyByWidget extends Widget {
   }
 
   _getCommMsgHandler() {
-    return (msg: KernelMessage.ICommMsgMsg) => {
+    return async (msg: KernelMessage.ICommMsgMsg) => {
       const msgContent: JSONValue = msg.content.data;
       switch ((msgContent as JSONObject).type) {
         case PYFLYBY_COMMS.MISSING_IMPORTS: {
           const itd = msgContent['missing_imports'];
-          this._insertImport(itd).then(imports => {
+          this._insertImport(itd).then(async imports => {
+            // Acquire new lock but wait for previous lock to expire
+            const currentLockId = this._lock.requestedLockCount;
+            this._lock.enable(currentLockId + 1);
+            await this._lock.promise[currentLockId];
```

Comment:
In case of a bug, maybe we have a timeout where we just send it anyway?

Why did we need sequential?

Reply:
When some code is run, pyflyby sends a missing import message containing the import that was missing. 
Then
jupyterlab_pyflyby replies with a format import message containing the code in the current import cell and the missing import
Then 
it receives a reply with the finally formatted cell (for a particular import) and replaces the content.

If it's not sequential there are cases where many of the format import messages read a stale/not updated version of the import cell due to which some imports are ignored.

Since there's no way to know how many missing import messages may come in, we need to handle them sequentially.

Reply:
I've a added a timeout for all locks. 1.5 sec seemed like a good value in practice. As a future enhancement we can allow users to configure that timeout value using a settings option.